splunk-otel-android/src/main/java/com/splunk/rum/RumInitializer.java

```diff
@@ -286,45 +291,88 @@ private Resource createSplunkResource() {
-                    initializationEvents.emit("slowRenderingDetectorInitialized");
+                    if (applicationId != null && !applicationId.isEmpty()) {
+                        builder.addAttributesExtractor(constant(APPLICATION_ID_KEY, applicationId));
+                    }
+                    if (versionCode != null && !versionCode.isEmpty()) {
+                        builder.addAttributesExtractor(constant(APP_VERSION_CODE_KEY, versionCode));
+                    }
+                    if (uuid != null && !uuid.isEmpty()) {
+                        builder.addAttributesExtractor(constant(SPLUNK_OLLY_UUID_KEY, uuid));
+                    }
+
+                    builder.setMainLooper(mainLooper).build().installOn(instrumentedApplication);
+
+                    initializationEvents.emit("anrMonitorInitialized");
                 });
     }
 
     private void installCrashReporter(OpenTelemetryRumBuilder otelRumBuilder) {
         otelRumBuilder.addInstrumentation(
                 instrumentedApplication -> {
-                    CrashReporter.builder()
-                            .addAttributesExtractor(
+                    String applicationId = null;
+                    String versionCode = null;
+                    String uuid = null;
+                    ErrorIdentifierExtractor extractor =
+                            ErrorIdentifierExtractor.getInstance(application);
+
+                    if (extractor != null) {
+                        applicationId = extractor.getApplicationId();
+                        versionCode = extractor.getVersionCode();
+                        uuid = extractor.getCustomUUID();
+                    }
+
+                    CrashReporterBuilder builder = CrashReporter.builder();
+                    builder.addAttributesExtractor(
                                     RuntimeDetailsExtractor.create(
                                             instrumentedApplication
                                                     .getApplication()
                                                     .getApplicationContext()))
-                            .addAttributesExtractor(new CrashComponentExtractor())
-                            .build()
-                            .installOn(instrumentedApplication);
+                            .addAttributesExtractor(new CrashComponentExtractor());
+
+                    if (applicationId != null && !applicationId.isEmpty()) {
+                        builder.addAttributesExtractor(constant(APPLICATION_ID_KEY, applicationId));
+                    }
+                    if (versionCode != null && !versionCode.isEmpty()) {
+                        builder.addAttributesExtractor(constant(APP_VERSION_CODE_KEY, versionCode));
+                    }
+                    if (uuid != null && !uuid.isEmpty()) {
+                        builder.addAttributesExtractor(constant(SPLUNK_OLLY_UUID_KEY, uuid));
+                    }
```

Comment:
So if the `ErrorIdentifierExtractor` were to implement `AttributesExtractor` AND not be nullable, then this could just read something like
```java
.addAttributesExtractor(new CrashComponentExtractor())
.addAttributesExtractor(new ErrorIdentifierExtractor(application))
```
...which I think is considerably cleaner and follows the existing/established patterns better.

Reply:
I would have liked to do that, would be much cleaner. However we can’t add these as attribute extractors. 

CrashReporterBuilder’s addAttributesExtractor takes in an AttributeExtractor that takes an objectCrashDetails object.
ANRDetectorBuilder’s addAttributeExtractor takes in an AttributeExtractor that takes a StackTraceElements object

To implement this as an attributesExtractor we’d have to pass in the Application object. And then add an additional addAttributesExtractor method to both builder classes. This stuff is in the otel android repo

Down the line when this is pushed upstream in otel android, we can add those changes to the otel android repo and then streamline this part. But by then we really can just add all this straight to otel android.

The nullable annotations are a bit verbose. But just trying to catch all edge cases in case there are any issues fetching any of the classes that we use to look up these fields.